### DISCUSSION TITLE
post/azure-blob-sas-guidelines

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#83] [Announcements]
# post/azure-blob-sas-guidelines

Mark Heath's Development Blog

https://markheath.net/post/azure-blob-sas-guidelines

**Comment by SpeedOfSpin** [comment]
We are just starting to implement SAS tokens and we generate on the fly short lived tokens. What doesn't feel right is that when we do an upsert back to the DB the URL if the image has the SAS tokens on it and we remove it before storing which seems a bit of a hack. Do you have any advice. Great article.

**Reply by markheath** [reply]
Yes, I would avoid storing SAS tokens in the database if you are able to generate them yourself, and generate them just in time. Only exception to that might be if you are needing to rapidly generate SAS tokens for the same blob many times in a short period in which case caching them for a while would make sense.
